src/Network/PushNotify/APN.hs

```diff
@@ -604,11 +610,16 @@ sendSilentMessage s token mJwt = catchErrors $
         sendApnRaw c token mJwt message
   where message = "{\"aps\":{\"content-available\":1}}"
 
-ensureOpen :: ApnSession -> IO ()
-ensureOpen s = do
-    open <- isOpen s
+ensureSessionOpen :: ApnSession -> IO ()
+ensureSessionOpen s = do
+    open <- isSessionOpen s
     unless open $ error "Session is closed"
 
+ensureConnectionOpen :: ApnConnection -> IO ()
+ensureConnectionOpen c = do
+    open <- isConnectionOpen c
+    unless open $ error "Connection is closed"
```

Comment:
I'm curious about this `error` right here. The context is that we're calling this in an `ExceptT . (try :: IO a -> IO (Either ClientError a))`. Which means that it *seems* like we're intending to have a `ClientError` as the "known" error path.

I know the other `ensureSessionIsOpen` also uses `error` - may be worthwhile to change that too? Would be a major break to change exception types IMO.